src/components/ProductList/ProductList.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import { useAllSearch } from '../../utils/hooks/useAllProducts';
+import PaginatedResults from './PaginationControl/Pagination';
+import { ImageGrid, ProductListMain, Spiner } from './ProductList.styles';
+import AsideBarComponent from './AsideBar';
+import { createCards, handleFilters, handleResults } from './ProductList.helper';
+import { useLocation } from "react-router-dom";
+
+
+export default function ProductList() {
+
+  const [filters, setFilters] = useState([]);
+  const [isFiltered, setIsFiltered] = useState(false);
+  const [results, setResults] = useState([]);
+  const location = useLocation();
+
+  let { data, isLoading } = useAllSearch();
+
+  const categoryParam = new URLSearchParams(location.search).get('category');
+  
+  useEffect(() => {
+    if (categoryParam && data?.results) {
+      const decodedParam = categoryParam.replaceAll(' & ', '--')
+      
+      const filtersSet = handleFilters(decodedParam, filters, setIsFiltered, setFilters);
+      const cards = handleResults(filtersSet, data);
+
+      setResults(cards);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data])
+  
+
+  const allProducts = data?.results;
+
+  const handleClickAsideBar = (value) => {
+    const filterName = value.target.innerText.replaceAll(' & ', '--');
+    const filtersSet = handleFilters(filterName, filters, setIsFiltered, setFilters);
+    const cards = handleResults(filtersSet, data);
+    
+    setResults(cards);
+  }
+
+  function clearFilterClick() {
+    setFilters([]);
+    setIsFiltered(false)
+  }
+  
+  const myProducts = createCards(allProducts);
+
+  return (
+    <ProductListMain>
+      <AsideBarComponent 
+        filters={filters}
+        handleClickAsideBar={item => handleClickAsideBar(item)}
+        clearFilterClick={clearFilterClick}
+      />
+      <ImageGrid>
+        {isLoading
+          ? <Spiner />
+          : <PaginatedResults size='12' data={isFiltered
```

Comment:
for this data we could perhaps utilize the `useEffect` hook, just so we don't nest ternary statements.